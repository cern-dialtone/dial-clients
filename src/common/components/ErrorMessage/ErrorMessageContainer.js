import { connect } from 'react-redux';
import withPhoneService from 'calls/providers/PhoneProvider/PhoneService';
import ErrorMessage from 'common/components/ErrorMessage/ErrorMessage';

function mapStateToProps({ call, connection, numbers, auth }) {
  return {
    errors: [call.error, connection.error, numbers.error, auth.error]
  };
}

export const ConnectedErrorMessageContainer = connect(
  mapStateToProps,
  null
)(ErrorMessage);

export default withPhoneService(ConnectedErrorMessageContainer);
