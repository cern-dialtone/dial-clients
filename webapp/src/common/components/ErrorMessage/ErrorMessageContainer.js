import { connect } from "react-redux";
import { phoneService } from "calls/providers/PhoneProvider/PhoneProvider";
import ErrorMessage from "common/components/ErrorMessage/ErrorMessage";

function mapStateToProps({ calls, auth }) {
  return {
    errors: [
      calls.call.error,
      calls.connection.error,
      calls.numbers.error,
      auth.error
    ]
  };
}

export const ErrorMessageContainer = connect(
  mapStateToProps,
  null
)(ErrorMessage);

export default phoneService(ErrorMessageContainer);
