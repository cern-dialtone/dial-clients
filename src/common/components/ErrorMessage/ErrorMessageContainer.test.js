import React from 'react';
import { render, waitForElement } from '@testing-library/react';

import { Provider } from 'react-redux';

import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import { createHashHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

import ErrorMessageContainer from './ErrorMessageContainer';

const history = createHashHistory();
const middlewares = [thunk, routerMiddleware(history)];
const mockStore = configureMockStore(middlewares);

describe('ErrorMessageContainer test', () => {
  let store;
  let initialState;

  beforeEach(() => {
    initialState = {
      call: {
        error: undefined
      },
      connection: {
        error: undefined
      },
      numbers: {
        error: undefined
      },
      auth: {
        error: undefined
      }
    };
    store = mockStore(initialState);
    store.dispatch = jest.fn();
  });

  it('renders component and includes undefined store props', async () => {
    const { container, asFragment } = render(
      <Provider store={store}>
        <ErrorMessageContainer t={key => key} />
      </Provider>
    );
    expect(container.firstChild).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders component and includes call error on store props', async () => {
    initialState.call.error = { message: 'This is error 1', statusCode: '1' };

    const { getByText, asFragment } = render(
      <Provider store={store}>
        <ErrorMessageContainer t={key => key} />
      </Provider>
    );
    await waitForElement(() => getByText('This is error 1 (1)'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders component and includes connection error on store props', async () => {
    initialState.connection.error = {
      message: 'This is error 1',
      statusCode: '1'
    };

    const { getByText, asFragment } = render(
      <Provider store={store}>
        <ErrorMessageContainer t={key => key} />
      </Provider>
    );
    await waitForElement(() => getByText('This is error 1 (1)'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders component and includes numbers error on store props', async () => {
    initialState.numbers.error = {
      message: 'This is error 1',
      statusCode: '1'
    };

    const { getByText, asFragment } = render(
      <Provider store={store}>
        <ErrorMessageContainer t={key => key} />
      </Provider>
    );
    await waitForElement(() => getByText('This is error 1 (1)'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders component and includes auth error on store props', async () => {
    initialState.auth.error = {
      message: 'This is error 1',
      statusCode: '1'
    };

    const { getByText, asFragment } = render(
      <Provider store={store}>
        <ErrorMessageContainer t={key => key} />
      </Provider>
    );
    await waitForElement(() => getByText('This is error 1 (1)'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders component and includes all errors in props', async () => {
    initialState.call.error = {
      message: 'This is error 1',
      statusCode: '1'
    };
    initialState.connection.error = {
      message: 'This is error 2',
      statusCode: '2'
    };
    initialState.numbers.error = {
      message: 'This is error 3',
      statusCode: '3'
    };
    initialState.auth.error = {
      message: 'This is error 4',
      statusCode: '4'
    };

    const { getByText, asFragment } = render(
      <Provider store={store}>
        <ErrorMessageContainer t={key => key} />
      </Provider>
    );
    await waitForElement(() => getByText('This is error 1 (1)'));
    await waitForElement(() => getByText('This is error 2 (2)'));
    await waitForElement(() => getByText('This is error 3 (3)'));
    await waitForElement(() => getByText('This is error 4 (4)'));

    expect(asFragment()).toMatchSnapshot();
  });
});
