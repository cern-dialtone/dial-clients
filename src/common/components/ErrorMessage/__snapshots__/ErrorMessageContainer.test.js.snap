// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessageContainer test renders component and includes all errors in props 1`] = `
<DocumentFragment>
  <div
    class="ui error message"
  >
    <i
      aria-hidden="true"
      class="close icon"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        An error has occurred
      </div>
      <ul
        class="list"
      >
        <li
          class="content"
        >
          This is error 1 (1)
        </li>
        <li
          class="content"
        >
          This is error 2 (2)
        </li>
        <li
          class="content"
        >
          This is error 3 (3)
        </li>
        <li
          class="content"
        >
          This is error 4 (4)
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ErrorMessageContainer test renders component and includes auth error on store props 1`] = `
<DocumentFragment>
  <div
    class="ui error message"
  >
    <i
      aria-hidden="true"
      class="close icon"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        This is error 1 (1)
      </div>
      <ul
        class="list"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ErrorMessageContainer test renders component and includes call error on store props 1`] = `
<DocumentFragment>
  <div
    class="ui error message"
  >
    <i
      aria-hidden="true"
      class="close icon"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        This is error 1 (1)
      </div>
      <ul
        class="list"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ErrorMessageContainer test renders component and includes connection error on store props 1`] = `
<DocumentFragment>
  <div
    class="ui error message"
  >
    <i
      aria-hidden="true"
      class="close icon"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        This is error 1 (1)
      </div>
      <ul
        class="list"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ErrorMessageContainer test renders component and includes numbers error on store props 1`] = `
<DocumentFragment>
  <div
    class="ui error message"
  >
    <i
      aria-hidden="true"
      class="close icon"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        This is error 1 (1)
      </div>
      <ul
        class="list"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ErrorMessageContainer test renders component and includes undefined store props 1`] = `<DocumentFragment />`;
